backend/de.metas.manufacturing/src/main/java/de/metas/manufacturing/generatedcomponents/PasswordGenerator.java

```diff
@@ -35,16 +43,70 @@
  * Borrowed from: https://mkyong.com/java/java-password-generator-example/
  */
 @Service
-public class PasswordGenerator
+public class PasswordGenerator implements IComponentGenerator
 {
 	private static final String CHAR_LOWERCASE = "abcdefghijklmnopqrstuvwxyz";
 	private static final String CHAR_UPPERCASE = CHAR_LOWERCASE.toUpperCase();
 	private static final String DIGIT = "0123456789";
 	private static final String PUNCTUATION = "!@#&()–[{}]:;',?/*";
 
+	/*package*/ static final String PARAM_LENGTH = "length";
+	/*package*/ static final String PARAM_USE_LOWERCASE = "useLowercase";
+	/*package*/ static final String PARAM_USE_UPPERCASE = "useUppercase";
+	/*package*/ static final String PARAM_USE_DIGIT = "useDigit";
+	/*package*/ static final String PARAM_USE_PUNCTUATION = "usePunctuation";
```

Comment:
instead of `/*package*/` which is not relevant info, pls annotate those constants with `@VisibleForTesting` because that's the reason for making them package level

Reply:
true